builder/Dockerfile

```diff
@@ -15,7 +15,7 @@ RUN dnf -y update \
         bison \
         ca-certificates \
         clang \
-        cmake \
+        cmake-3.18.2-9.el8 \
```

Comment:
Looks like the default version for cmake on centos stream8 is `3.20.2`, why do we need to downgrade it? Maybe we could add a comment on why this is needed and whether there's something we could do to go back to using the latest version?

Also, we need to double check the downstream build.

Reply:
This is an awkward one because that version of cmake was segfaulting when configuring the project, but pinning to the previous available version (as in this change) prevented the segfault.

Reply:
That is... Not good.

We should probably not sleep on this and take a closer look at what's going on, but for now we can probably go ahead with the merge.